### ISSUE TITLE
Custom filters dated older than the default don't work

### ISSUE THREAD

**Issue by nityy** [bromite/bromite#1329]
### Bromite version

Version: `92.0.4515.134`
Arch: `arm64`
Android version: `11.0`
Device model: `XT1804`

### Flags

These are the flags changed under `chrome://flags`:
- None changed

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No

### Is this bug happening in an incognito tab?
No, in a regular tab but It happens in an incognito tab as well.

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?

No

### Describe the bug

If you try a custom filter with a newer (e.g. generated today) `Last-Modified` header, then you can't switch to a different custom filter which has older header, or even the bromite's default because there's a good chance it is few days old. It gets stuck at that filter even if URL has been changed. You either wait for the filter you want to use to receive an update or clear data of browser. 

### Steps to reproduce the bug

1. Assuming default filter is set, go to [this adblock testing page](https://d3ward.github.io/toolz/src/adblock) and notice that it says 100% on top and everything (except apple which is irrelevant) is green i.e. blocked
2. Set [this link](https://github.com/nityy/filtrite/releases/latest/download/test-lucky.dat) as the filter URL in Adblock settings. This filter blocks only five domains of Luckyorange (Analytics) on that testing page
3. Restart the browser and wait for 5 seconds, go to the testing page and verify the new list is working i.e. only Luckyorange is in green
4. Set the filter URL back to the [default](https://www.bromite.org/filters/filters.dat)
5. Restart the browser and wait for 5 seconds, go to the testing page and notice that still only Luckyorange is in green

### Expected behavior
At step 5, everything should be blocked on that page, just like what we started with.

Checking for this header only makes sense if the URL is the same. Skip `Last-Modified` header check, if the filter URL has changed

**Comment by csagan5** [comment]
We only have information about the timestamp available and I am not going to re-engineer the feature to fix this bug; the bug won't fix, #811 will provide a workaround/mitigation.
